sklearn/neighbors/nearest_centroid.py

```diff
@@ -133,6 +133,8 @@ def fit(self, X, y):
                     self.centroids_[cur_class] = np.median(X[center_mask], axis=0)
                 else:
                     self.centroids_[cur_class] = csc_median_axis_0(X[center_mask])
+            elif self.metric == 'precomputed':
```

Comment:
I guess so. And perhaps the error message should be more informative and explicitly say precomputed not supported.